test/e2e/route_metrics_test.go

```diff
@@ -348,11 +410,13 @@ func waitForRouteMetricsAddorUpdate(t *testing.T, prometheusClient prometheusv1.
 
 		// Check if length of returned metric Vector is zero.
 		if len(vec) == 0 {
+			t.Logf("metric vector is empty...retrying")
 			return false, nil
 		}
 
 		// Check if metrics is updated.
 		if !vec[0].Value.Equal(model.SampleValue(float64(value))) {
+			t.Logf("expected metric value %d, but got %d...retrying", value, int(vec[0].Value))
```

Comment:
FYI this can be a little annoying, but I suppose better than nothing:
```
    route_metrics_test.go:397: waiting for route_metrics_controller_routes_per_shard{shard_name=ic-rs-ns-metrics-test} to become 0
    route_metrics_test.go:419: expected metric value 0, but got 1...retrying
    route_metrics_test.go:419: expected metric value 0, but got 1...retrying
    route_metrics_test.go:419: expected metric value 0, but got 1...retrying
    route_metrics_test.go:419: expected metric value 0, but got 1...retrying
    route_metrics_test.go:419: expected metric value 0, but got 1...retrying
    route_metrics_test.go:419: expected metric value 0, but got 1...retrying
    route_metrics_test.go:419: expected metric value 0, but got 1...retrying
    route_metrics_test.go:419: expected metric value 0, but got 1...retrying
    route_metrics_test.go:419: expected metric value 0, but got 1...retrying
    route_metrics_test.go:419: expected metric value 0, but got 1...retrying
    route_metrics_test.go:419: expected metric value 0, but got 1...retrying
    route_metrics_test.go:419: expected metric value 0, but got 1...retrying
    route_metrics_test.go:419: expected metric value 0, but got 1...retrying
    route_metrics_test.go:419: expected metric value 0, but got 1...retrying
    route_metrics_test.go:419: expected metric value 0, but got 1...retrying
    route_metrics_test.go:419: expected metric value 0, but got 1...retrying
    route_metrics_test.go:419: expected metric value 0, but got 1...retrying
    route_metrics_test.go:419: expected metric value 0, but got 1...retrying
    route_metrics_test.go:419: expected metric value 0, but got 1...retrying
    route_metrics_test.go:419: expected metric value 0, but got 1...retrying
    route_metrics_test.go:419: expected metric value 0, but got 1...retrying
    route_metrics_test.go:419: expected metric value 0, but got 1...retrying
    route_metrics_test.go:419: expected metric value 0, but got 1...retrying
    route_metrics_test.go:419: expected metric value 0, but got 1...retrying
    route_metrics_test.go:419: expected metric value 0, but got 1...retrying
    route_metrics_test.go:419: expected metric value 0, but got 1...retrying
    route_metrics_test.go:419: expected metric value 0, but got 1...retrying
    route_metrics_test.go:419: expected metric value 0, but got 1...retrying
    route_metrics_test.go:419: expected metric value 0, but got 1...retrying
    route_metrics_test.go:419: expected metric value 0, but got 1...retrying
    route_metrics_test.go:397: waiting for route_metrics_controller_routes_per_shard{shard_name=ic-rs-ns-metrics-test} to become 1
```